exp/api/v1alpha3/types.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1alpha3"
+)
+
+type (
```

Comment:
Should these types live elsewhere since they are not referenced in MachinePool or MachinePoolStatus?

Reply:
I would be happy to move these. Again, this was following the idiom set in the code base thus far.

We have discussed a refactor of this pattern, but I'm hesitant to set the precedent in this PR.

https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/6a820e03f2ca57f54ce3f2845cc5e0a99fb41b87/api/v1alpha3/azuremachine_types.go#L32-L62

Reply:
I would be happy to move these. Again, this was following the idiom set in the code base thus far.

We have discussed a refactor of this pattern, but I'm hesitant to set the precedent in this PR.

https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/6a820e03f2ca57f54ce3f2845cc5e0a99fb41b87/api/v1alpha3/types.go#L297-L331

Reply:
Let's leave it for now.